src/overlays/actors/ovl_Bg_Goron_Oyu/z_bg_goron_oyu.c

```diff
@@ -28,31 +32,162 @@ const ActorInit Bg_Goron_Oyu_InitVars = {
     (ActorFunc)BgGoronOyu_Draw,
 };
 
-#endif
-
-extern UNK_TYPE D_06000968;
-extern UNK_TYPE D_06000988;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Goron_Oyu/func_80B40080.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Goron_Oyu/func_80B4009C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Goron_Oyu/func_80B400C8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Goron_Oyu/func_80B40100.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Goron_Oyu/func_80B40160.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Goron_Oyu/func_80B401F8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Goron_Oyu/func_80B40308.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Goron_Oyu/func_80B40394.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Goron_Oyu/BgGoronOyu_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Goron_Oyu/BgGoronOyu_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Goron_Oyu/BgGoronOyu_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Goron_Oyu/BgGoronOyu_Draw.s")
+extern Gfx D_06000080;
+extern Gfx D_06000158;
+extern AnimatedMaterial D_06000968;
+extern CollisionHeader D_06000988;
+
+void func_80B40080(BgGoronOyu* this) {
+    this->unk_17E = 1;
+    this->actionFunc = func_80B400C8;
+}
+
+void func_80B4009C(BgGoronOyu* this) {
+    this->unk_17E = 0;
+    this->initialActorCutscene = this->dyna.actor.cutscene;
+    this->actionFunc = func_80B40100;
+    this->unk_164 = 20.0f;
+}
+
+void func_80B400C8(BgGoronOyu* this, GlobalContext* globalCtx) {
+    BgGoronOyu_UpdateWaterBoxInfo(this, globalCtx);
+    func_80B401F8(this, globalCtx);
+}
+
+void func_80B40100(BgGoronOyu* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCanPlayNext(this->initialActorCutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->initialActorCutscene, &this->dyna.actor);
+        this->actionFunc = func_80B40160;
+    } else {
+        ActorCutscene_SetIntentToPlay(this->initialActorCutscene);
+    }
+}
+
+void func_80B40160(BgGoronOyu* this, GlobalContext* globalCtx) {
+    static Vec3f D_80B40780 = { 0, 0, 0 };
+    Math_StepToF(&this->unk_164, 0.0f, 0.2f);
```

Comment:
Please add an empty line between the variables declarations and the actual code
```suggestion
    static Vec3f D_80B40780 = { 0, 0, 0 };

    Math_StepToF(&this->unk_164, 0.0f, 0.2f);
```